lib/src/charts/data/core/cartesian/cartesian_data.dart

```diff
@@ -23,28 +20,61 @@ enum CartesianChartAlignment {
 enum CartesianChartOrientation { vertical, horizontal }
 
 abstract class CartesianSeries {
-  T when<T>({required T Function() barSeries}) {
+  T when<T>({
+    required T Function(BarSeries series) onBarSeries,
+  }) {
+    switch (runtimeType) {
+      case BarSeries:
+        return onBarSeries(this as BarSeries);
+      default:
+        throw TypeError();
+    }
+  }
+
+  T maybeWhen<T>({
+    T Function(BarSeries series)? onBarSeries,
+    required T Function() orElse,
+  }) {
     switch (runtimeType) {
       case BarSeries:
-        return barSeries();
+        return onBarSeries?.call(this as BarSeries) ?? orElse();
+      default:
+        throw TypeError();
+    }
+  }
+
+  static T whenSeries<T>(
+    Type type, {
+    T Function()? onBarSeries,
+    required T Function() orElse,
+  }) {
+    switch (type) {
+      case BarSeries:
+        return onBarSeries?.call() ?? orElse();
       default:
         throw TypeError();
     }
   }
 }
 
+abstract class CartesianConfig {}
+
 List<Tween<CartesianSeries>>? toCartesianTweens(
   List<CartesianSeries>? current,
   List<CartesianSeries> target,
 ) {
   return buildTweens(current, target, builder: (current, target) {
     final currentValue =
         current == null || current.runtimeType != target.runtimeType
-            ? BarSeries.zero()
+            ? null
             : current;
     return target.when(
-      barSeries: () => BarSeriesTween(
-          begin: currentValue as BarSeries, end: target as BarSeries),
+      onBarSeries: (series) {
+        return BarSeriesTween(
+          begin: currentValue.ifNullThen(BarSeries.zero()),
```

Comment:
`ifNullThen` is confusing. 
It's not just checking for null, it's internally calling `asOrNull`.
Can we rename this to `asOrDefault`?

Reply:
renamed to `asOrDefault`